src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -138,21 +138,21 @@ theorem ofBool_eq_iff_eq : ∀(b b' : Bool), BitVec.ofBool b = BitVec.ofBool b'
   getLsb (x#'lt) i = x.testBit i := by
   simp [getLsb, BitVec.ofNatLt]
 
-@[simp, bv_toNat] theorem toNat_ofNat (x w : Nat) : (x#w).toNat = x % 2^w := by
+@[simp, bv_toNat] theorem toNat_ofNat (x w : Nat) : (BitVec.ofNat w x).toNat = x % 2^w := by
   simp [BitVec.toNat, BitVec.ofNat, Fin.ofNat']
 
 -- Remark: we don't use `[simp]` here because simproc` subsumes it for literals.
 -- If `x` and `n` are not literals, applying this theorem eagerly may not be a good idea.
 theorem getLsb_ofNat (n : Nat) (x : Nat) (i : Nat) :
-  getLsb (x#n) i = (i < n && x.testBit i) := by
+  getLsb (BitVec.ofNat n x) i = (i < n && x.testBit i) := by
   simp [getLsb, BitVec.ofNat, Fin.val_ofNat']
 
 @[simp, deprecated toNat_ofNat (since := "2024-02-22")]
-theorem toNat_zero (n : Nat) : (0#n).toNat = 0 := by trivial
+theorem toNat_zero (n : Nat) : (BitVec.ofNat n 0).toNat = 0 := by trivial
```

Comment:
Are these changes intentional? It seems fine to leave the notation in place for literals!

I can change them back if you like.

Reply:
@semorrison Are you suggesting we use the following variant?
```
syntax:max num noWs "#" noWs term:max : term
```
That is, the bitvector size can still be a term. If it does not conflict with Mathlib notation, I am happy with the change, and I think it is a good one.